hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3972,11 +3973,91 @@ func (r *HostedClusterReconciler) validateHostedClusterSupport(hc *hyperv1.Hoste
 }
 
 func (r *HostedClusterReconciler) validateNetworks(hc *hyperv1.HostedCluster) error {
-	errs := validateSliceNetworkCIDRs(hc)
+	var errs field.ErrorList
+	errs = append(errs, validateSliceNetworkCIDRs(hc)...)
+	errs = append(errs, checkAdvertiseAddressOverlapping(hc)...)
 
 	return errs.ToAggregate()
 }
 
+// findAdvertiseAddress function returns a string indicating the AdvertiseAddress for the hostedcluster.
+// if the advertise address is properly set, it will return that value, in any other case,
+// it will set the defaultIPv4 or defaultIPv6 advertise address depending on the
+// ClusterNetwork Primary IP.
+func findAdvertiseAddress(hc *hyperv1.HostedCluster) string {
+	var advertiseAddress *string
+	if hc.Spec.Networking.APIServer != nil && hc.Spec.Networking.APIServer.AdvertiseAddress != nil {
+		advertiseAddress = hc.Spec.Networking.APIServer.AdvertiseAddress
+
+		ipaddr := net.ParseIP(*advertiseAddress)
+		if ipaddr != nil {
+			return *advertiseAddress
+		}
+	}
+
+	ipaddr := net.ParseIP(hc.Spec.Networking.ClusterNetwork[0].CIDR.IP.String())
```

Comment:
why do we omit the case where the input is not parseable? As a user I'd expect we fail in such a case rather than use default

Reply:
Addressed